src/Features/Core/Portable/CodeLens/CodeLensHelpers.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.CodeLens;
+
+internal static class CodeLensHelpers
+{
+    public static DocumentId? GetSourceGeneratorDocumentId(IDictionary<object, object> descriptorProperties)
```

Comment:
moved from later code.  used by us, and then also exposed through EA for UnitTesting to use.